### PR TITLE
Added guide in UI to help users write queries when searching Posts

### LINKED ISSUE #1276: Add guide in UI to help users write queries when searching Posts
**What would you like to be added**:
Add some type of guide in UI to help users in writing more complex searches on this screen:

image

**Why would you like this to be added**:
#1275 will allow users to use operators + other options to fine-tune their searches:

image

This type of syntax might not be obvious to most people, we should create some type of guide to help users in using the new syntax. Please refer to that PR for more information on writing search queries.

### PR DESCRIPTION
**Pull request by MLJBrackett**
## Issue This PR Addresses
Fixes #1276

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description
Adds a ButtonIcon in the SearchBar component that allows users to hover over and see how to properly query while searching for posts. This pull request is related to #1275 [Image from Gyazo](https://gyazo.com/e551919ced5612adb5639e91625b786a)

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
Test Suites: 26 passed, 26 total
Tests:       165 passed, 165 total
Snapshots:   0 total
Time:        11.03 s
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by c3ho** [comment]
I like that is not intrusive. With that being said there's also #1225 that is going to reposition the search button @humphd how should we handle this?

**Comment by humphd** [comment]
I'd move it up beside the "Search" title at the top.

**Review by c3ho — CHANGES_REQUESTED** [review]
Please see @humphd 's comment

**Comment by MLJBrackett** [comment]
@c3ho @humphd Fixed, this is what it looks like now.

[Image from Gyazo](https://gyazo.com/7c68b6fba771bc0938c35af48e4f91c1)

**Comment by c3ho** [comment]
Nice, I hesitate to say this as I'm extremely bad at design... could we remove the blue background for the button and just keep the white? Blue on blue seems rough

File: src/frontend/src/components/SearchBar/SearchBar.jsx [hunk 1]
```diff
@@ -114,11 +130,59 @@ function CustomizedInputBase(props) {
           alignItems="center"
           justify="flex-start"
         >
-          <Grid item xs={12}>
+          <Grid item>
             <Typography variant="h1" className={classes.h1}>
               Search
             </Typography>
           </Grid>
+          <HtmlTooltip
```
Comment (humphd) [inline src/frontend/src/components/SearchBar/SearchBar.jsx:138]: Let's break this out into its own component, `SearchHelp.jsx` so it's not so massive in this file.  You should be able to use it here with `<SearchHelp />`

**Review by humphd — CHANGES_REQUESTED** [review]
Let's refactor this, and can you change it to not be a button (i.e., we can't click it), it's just an icon you can hover:

Screen Shot 2020-11-17 at 1 47 26 PM

**Comment by MLJBrackett** [comment]
Just fixed it check it out now, I think it looks a lot better without the button format as well.

File: src/frontend/src/components/SearchBar/SearchBar.jsx [hunk 2]
```diff
@@ -60,16 +61,16 @@ const useStyles = makeStyles((theme) => ({
       backgroundColor: theme.palette.secondary.dark,
     },
     '& * > .MuiSvgIcon-root': {
-      fontSize: '2rem',
+      fontSize: '3rem',
       color: theme.palette.text.primary,
     },
     margin: 0,
+    marginLeft: 15,
```
Comment (humphd) [inline src/frontend/src/components/SearchBar/SearchBar.jsx:68]: 15 what?  Needs a unit.

File: src/frontend/src/components/SearchHelp/SearchHelp.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    border: '1px solid #dadde9',
+  },
+}))(Tooltip);
+
+function SearchHelp(props) {
+  const classes = useStyles();
+  return (
+    <HtmlTooltip
+      title={
+        <React.Fragment>
```
Comment (humphd) [inline src/frontend/src/components/SearchHelp/SearchHelp.jsx:31]: Can you import Fragment at the top vs. here?

Comment (humphd) [reply src/frontend/src/components/SearchHelp/SearchHelp.jsx:31]: Or just use`<>...</>`

**Review by humphd — CHANGES_REQUESTED** [review]
This needs a rebase to pickup the new changes in SearchBar.jsx.  Can you do that and I'll re-review.

File: src/frontend/src/components/SearchBar/SearchBar.jsx (hunk 2 above)
Comment (MLJBrackett) [reply src/frontend/src/components/SearchBar/SearchBar.jsx:68]: I need to remove all changes to the SearchBar for that kind of stuff actually.

**Comment by MLJBrackett** [comment]
@humphd I've messaged you on slack about an issue I am having while rebasing my code, I don't think the issue has something to do with my branch though.

File: src/frontend/src/components/SearchBar/SearchBar.jsx [hunk 4]
```diff
@@ -103,11 +105,12 @@ function CustomizedInputBase(props) {
           alignItems="center"
           justify="flex-start"
         >
-          <Grid item xs={12}>
```
Comment (humphd) [inline src/frontend/src/components/SearchBar/SearchBar.jsx:106]: Why this change?

**Review by humphd — APPROVED** [review]

File: src/frontend/src/components/SearchBar/SearchBar.jsx (hunk 4 above)
Comment (MLJBrackett) [reply src/frontend/src/components/SearchBar/SearchBar.jsx:106]: Leaving that in wouldn't allow me to put the <SearchHelp> beside the header since it has built in css with xs={12}

Comment (humphd) [reply src/frontend/src/components/SearchBar/SearchBar.jsx:106]: OK, sounds good.

**Review by PedroFonsecaDEV — APPROVED** [review]
